### DISCUSSION TITLE
how to enable HMR?

### DISCUSSION THREAD

**Discussion by realyukii** [unocss/unocss#2622] [Q&A]
I try to disable webpack cache with `config.cache = false`, delete `node_modules` & `.next` directory, but HMR still not working, Is there something I'm missing?

nextjs version 13

package information:
```
"next-transpile-modules": "latest",
"react": "latest",
"react-dom": "latest",
"next": "latest",
"@unocss/webpack": "^0.51.12",
"unocss": "^0.51.12"
```

**Comment by levi-Ma** [comment]
https://unocss.dev/integrations/postcss

**Reply by mdxiaohu** [reply]
@levi-Ma The postcss plugin seems to be unable to use class .
I see that the document can only be operated using instructions .
For example: '@apply' .

**Comment by mdxiaohu** [comment]
Of course, the issue of abandoning unocss will be completely resolved .

**Reply by levi-Ma** [reply]
The “@apply” function is not used, and the amount of code is not that large

**Reply by mdxiaohu** [reply]
Unocss is really difficult to use on webpack .
Probably because the author uses vite, there is insufficient maintenance for other versions .

I have also tried using this .
`https://unocss.dev/integrations/postcss`
Its advantage in webpack is that it can be operated using instructions .
for example: `@apply text-red;`
But the following methods cannot be used .
`<div class="text-red">test</div>`
Another issue is that there are good times and bad times .
Sometimes it's possible to parse '@apply', but sometimes it's not possible.
I haven't found the reason yet .


Another solution .
`@unocss/webpack`
Its first issue is that HMR cannot be used .
After each class update, the interface needs to be refreshed to see the effect .
This greatly affects development efficiency .
The second issue is that the instruction plugin is invalid .
Even if configured according to the document or issue, it is useless .
for example: `@apply text-red;`
This will not render to the corresponding style .


I have been researching for several days and looking at relevant issues, but there are no good solutions available .
Using unocss suggests directly accessing vite .
Or we can just discard it .
Of course, if you don't mind the issues with webpack, you can still reluctantly use it. However, this will be a disaster film for the team .

The above are all issues that occurred after configuring Vuecli4 and cli5 according to the official documentation provided by me .

**Reply by zyyv** [reply]
Personally, I am a fan of Vite. I am not very interested in webpack. And PR welcome to maintain the webpack package.
As a workaround, you can try to use https://unocss.dev/integrations/cli and it also support `transformers`.
